plugins/services/src/js/components/forms/VolumesFormSection.tsx

```diff
@@ -198,24 +185,110 @@ class VolumesFormSection extends React.Component<{
     );
   }
 
-  getExternalVolumeConfig(volume, key) {
-    if (volume.type !== "EXTERNAL") {
-      return null;
-    }
+  getExternalCSIVolumeConfig(volume, key) {
+    const updateVolume = (v) => {
+      this.props.onChange({
+        target: { name: `volumes.${key}`, value: { ...volume, ...v } },
+      });
+    };
```

Comment:
here we send our "manual events". (i'll refer to that in later comments).

this is one part of the main trick for not having to implement a reducer-statement that's tied to the name of a native input. so we make things really explicit here.